pkgs/os-specific/linux/bpftrace/default.nix

```diff
@@ -36,10 +36,25 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-QDqHAEVM/XHCFMS0jMLdKJfDUOpkUqONOf8+Fbd5dCY=";
   };
 
-  # libbpf 0.6.0 relies on typeof in bpf/btf.h to pick the right version of
-  # btf_dump__new() but that's not valid c++.
-  # see https://github.com/iovisor/bpftrace/issues/2068
-  patches = [ ./btf-dump-new-0.6.0.patch ];
+  patches = [
+    # libbpf 0.6.0 relies on typeof in bpf/btf.h to pick the right version of
+    # btf_dump__new() but that's not valid c++.
+    # see https://github.com/iovisor/bpftrace/issues/2068
+    ./btf-dump-new-0.6.0.patch
+
+    # "Add executable permission to tools"
+    (fetchpatch {
+      url = "https://github.com/iovisor/bpftrace/commit/77e524e6d276216ed6a6e1984cf204418db07c78.patch";
+      sha256 = "sha256-p0+GaJvZb0aHzd/Q+xRnwO3i43gTAs1IzDbea3ZuAAk=";
+    })
+
+    # "tools: Allow skipping examples"
+    (fetchpatch {
+      url = "https://github.com/iovisor/bpftrace/pull/2256.patch";
```

Comment:
upstream has been reactive with your previous PRs, let's wait until this gets merged imo

Reply:
What's the gain?

Given the introduced option stays the same and nothing else gets committed to `doc/CMakeLists.txt` before my PR is merged, fetching `.../pull/2256.pach` would make no difference to `.../commit/<id>.patch` wrt. `sha256`, I believe.

Reply:
on the off chance the commit is refused upstream we'd need to revisite the solution here, it has had no feedback at all yet.

afaik we don't like to diverge from upstream, I personally think these commits don't fix anything urging so would normally say we can wait for 0.15.1 or 0.16 or whatever comes next, but I can agree the release cycle is slow enough so won't push back for merged changes. 

Unmerged changes OTOH should wait, especially since we already have a corresponding rm in the install phase that does exactly the same thing, it's just a cleanup commit not fixing any real problem.

Reply:
> on the off chance the commit is refused upstream we'd need to revisite the solution here, it has had no feedback at all yet.
> 
> afaik we don't like to diverge from upstream, I personally think these commits don't fix anything urging so would normally say we can wait for 0.15.1 or 0.16 or whatever comes next, but I can agree the release cycle is slow enough so won't push back for merged changes.
> 
> Unmerged changes OTOH should wait, especially since we already have a corresponding rm in the install phase that does exactly the same thing, it's just a cleanup commit not fixing any real problem.

Right, I could leave the `fetchpatch`es out and just add this to `postInstall`:
```
ln -s $out/share/bpftrace/tools/*.bt $out/bin/
chmod +x $out/bin/*.bt
```

But I am under the impression that pulling patches is preferred and snippets like the above should always go with comments, which in this case meant links to the PRs/patches I submitted... so my reasoning was:  why not just fetch them directly (documents itself)?

I'm fine with it either way.

Reply:
I think it's a bit more subtle than just "preferring patches".

I definitely agree with what's been said in the previous PR: an upstream patch is just easier to maintain. When I or someone does next upgrade we just need to remove the patch and not worry about other modifications to undo (even if in this case there's no real harm in leaving an extra chmod, it's still best not to add it in the first place)

If it's not merged upstream by the time we get it in nixpkgs though it's a lot more work: Next time I upgrade the package I'll have forgotten about this PR (yes, my memory is bad), so I'll need to go look up at either what introduced the fetchpatch or/and the upstream PR to check what is was about, if it was merged without modification or if I need to update something, etc.
As a fetch from PR url it also risks you updating your PR, which will change the hash and make bpftrace unbuildable in further rebuilds.

In this case the whole purpose of this patch is just to replace the `rm -rf $out/share/bpftrace/tools/doc` below which is already here -- just adding a comment it can likely be replaced next update by whatever switch the PR ends up adding is less work for everyone. But maybe I'm just lazy :)

Reply:
I just rebased and replaced the patches with extra `postInstall` lines plus comments/links.
This should be simpler and yield less nixpkgs churn.